shoop/admin/modules/orders/static_src/create/view/lines.js

```diff
@@ -6,87 +6,146 @@
  * This source code is licensed under the AGPLv3 license found in the
  * LICENSE file in the root directory of this source tree.
  */
-import {addLine, deleteLine, retrieveProductData, setLineProperty} from "../actions";
-import {selectBox, LINE_TYPES} from "./utils";
+import {addLine, deleteLine, retrieveProductData, setLineProperty, updateTotals} from "../actions";
+import {LINE_TYPES, selectBox} from "./utils";
 import BrowseAPI from "BrowseAPI";
 
-export function renderOrderLines(store, lines) {
+function renderNumberCell(store, line, value, fieldId, canEditPrice, min=null) {
+    return m("input.form-control", {
+            type: "number",
+            step: line.step,
+            min: min != null ? min : "",
+            value: value,
+            disabled: !canEditPrice,
+            onchange: function () {
+                store.dispatch(setLineProperty(line.id, fieldId, this.value));
+                if (fieldId === "quantity" && line.product) {
+                    store.dispatch(retrieveProductData(
+                        {id: line.product.id, forLine: line.id, quantity: this.value}
+                    ));
+                }
+                store.dispatch(updateTotals(store.getState));
+            }
+    });
+}
+
+export function renderOrderLines(store, shop, lines) {
     return _(lines).map((line) => {
         var text = line.text, canEditPrice = true;
         const showPrice = (line.type !== "text");
         if (line.type === "product") {
             text = m("a", {
-                href: "#",
-                onclick: () => {
-                    BrowseAPI.openBrowseWindow({
-                        kind: "product",
-                        onSelect: (obj) => {
-                            store.dispatch(setLineProperty(line.id, "product", obj));
-                            store.dispatch(retrieveProductData({id: obj.id, forLine: line.id}));
-                        }
-                    });
-                }
-            }, (line.product ? line.product.text : gettext("No product selected")));
+                    href: "#",
+                    onclick: () => {
+                        BrowseAPI.openBrowseWindow({
+                            kind: "product",
+                            shop: shop.id,
+                            onSelect: (obj) => {
+                                store.dispatch(setLineProperty(line.id, "product", obj));
+                                store.dispatch(retrieveProductData(
+                                    {id: obj.id, forLine: line.id, quantity: line.quantity}
+                                ));
+                            }
+                        });
+                    }
+                }, (line.product ?
+                    [line.product.text, m("br"), m("small", "(" + line.sku + ")")] : gettext("No product selected"))
```

Comment:
this could be a button with a text "select product" instead of "no product selected"